### ISSUE TITLE
ckfinder not working - nginx error

### ISSUE THREAD

**Issue by le-zell** [martialblog/docker-limesurvey#133]
``using nginx + docker 5.4.10 version

When accessing to "browse the server" to upload images for example.

The page on URI : <hostname>/third_party/kcfinder/browse.php?type=images&CKEditor=description_fr&CKEditorFuncNum=2&langCode=fr has missing js and css files witch are rendered by css/index.php and js/index.php files.

in logs we have nginx errors : 
```
rewrite or internal redirection cycle while internally redirecting to "index.php"
```

maybe an nginx error configuration

**Comment by martialblog** [comment]
Hi, could this be related to the issue #131 ? There also was some mentioning of the ckfinder plugin.

My first instinct is to thing about the BASE_URL and PUBLIC_URL parameters. However, there are also issues reported with the app not playing nicely with reverse proxies regardless of the parameters. Could be related. 

Regards

**Comment by le-zell** [comment]
Yes i think so.

It's maybe relative to a config parameter `''urlFormat' => 'path',` witch shortcuts the URIs...don't know...

as a workaround i hacked `third_party/kcfinder/tpl/tpl_browser.php` and put `<base href="/third_party/kcfinder/">` instead of `<base href="<?= Yii::app()->request->getBaseUrl() ?>/">` 

I get a more fonctionnal webpage (with css/js resources). But the module is not working better...

**Comment by martialblog** [comment]
There's probably a way to tell the Yii::app that it is running behind a proxy, so that the routing is handled correctly.

**Comment by bovender** [comment]
I cannot get it to work either.

The 500 error reported at the top of this issue seems to be an nginx server config issue. The nginx config files has a `location ~ \.php$ {...}` directive which contains the line `try_files $uri /index.php`. When I change this to `try_files $uri /index.php?$argsi`, the 500 error no longer occurs, but a 404 error occurs instead. And in my opinion this is the root of the problem:

Script `/vendor/kcfinder/browse.php` attempts to access the following files:

- `index.php`
- `css.php`
- `js_localize.php`
- `js.php`

All of these throw a 404 error, because `browse.php` references them in the _root_ folder (example.com/), but in fact they reside in `/vendor/kcfinder`. It's not so much about the server domain, it's the path prefix that is missing.

I see this with @martialblog's LimeSurvey 6.x as well as 5.x images.

What puzzles me is that the LimeSurvey image by @adamzammit does _not_ exhibit this behavior. I tried to find out what the differences are between the images by @martialblog and @adamzammit, but I was unable to see anything that might be relevant (e.g., different configuration or so).

This is what the 'ressources' page of a survey looks like in my browser with @martialblog 's image (the buttons are not functional):

![image](https://github.com/martialblog/docker-limesurvey/assets/1484759/ba50a301-d6f1-4ea9-af95-b0afaec546e6)

And this is a screenshot of LimeSurvey running in a container based on @adamzammit's image (buttons are functional):

![image](https://github.com/martialblog/docker-limesurvey/assets/1484759/681a533f-205e-4c0b-961e-1f5f0d7f2d0d)

To summarize, there seems to be a directory issue with the kcfinder plugin in this image.

**Comment by bovender** [comment]
P.S. Here's a `docker-compose.yml` file that helps to quickly try out the two different images:

<details><summary>(expand)</p>
</summary><p>

```yaml
version: '2'

services:

  adamzammit:
    image: adamzammit/limesurvey
    ports:
      - 127.0.0.1:9090:80
    environment:
      LIMESURVEY_DB_PASSWORD: example
      LIMESURVEY_ADMIN_USER: admin
      LIMESURVEY_ADMIN_PASSWORD: admin
      LIMESURVEY_ADMIN_NAME: Admin
      LIMESURVEY_ADMIN_EMAIL: admin@example.com
    depends_on:
      - mysql

  martialblog:
    image: docker.io/martialblog/limesurvey:latest
    environment:
      - DBENGINE=InnoDB
      - DB_TYPE=mysql
      - DB_HOST=mysql
      - DB_PASSWORD=example
      - DB_NAME=martialblog
      - DB_USERNAME=root
      - ADMIN_USER=admin
      - ADMIN_NAME=Admin
      - ADMIN_PASSWORD=admin
      - ADMIN_EMAIL=admin@example.com
      - PUBLIC_URL=foobar.com
    ports:
      - 127.0.0.1:9091:8080
    depends_on:
      - mysql

  mysql:
    image: mariadb
    environment:
      MYSQL_ROOT_PASSWORD: example
``` 

</p></details>

Start up these services in the usual way, invoke http://localhost:9090/admin and http://localhost:9091/admin, log in with admin/admin, quickly create new surveys, and go to the 'Resources' page to see whether ckfinder works or does not work.

**Comment by bovender** [comment]
I've never been able to figure out why one image works (with regard to this issue) and the other doesn't, despite diff'ing entire directory trees.

However, in case anyone else besides me experiences problems with this, I've forked @adamzammit's 
limesurvey-docker repository and added Postgres support. In this container, the resources page works. I've created a PR on the original repository: adamzammit/limesurvey-docker#51

Works well on my server.

**Comment by albion2790** [comment]
Hey there. Im still having the same issue with this. Using limesurvey 6.5 and i cant seem to browse the server. 
![image](https://github.com/martialblog/docker-limesurvey/assets/54589115/fd0def71-713f-48dd-ab08-6475b9da1817)
This happens when i try to upload an image but the whole thing is not responding and the buttons are not doing anything. Is there a potential fix for this? Or is there any justification on why this is happening?

**Comment by jonas-kiesewetter** [comment]
Issue still exists, unable to find a working solution. Have to use a other image :(

**Comment by bovender** [comment]
Adam Zammit's image works very well and since reporting this issue, we have introduced support for Postgres databases, if that is why you were using this one here. https://github.com/adamzammit/limesurvey-docker

**Comment by jonas-kiesewetter** [comment]
> Adam Zammit's image works very well and since reporting this issue, we have introduced support for Postgres databases, if that is why you were using this one here. https://github.com/adamzammit/limesurvey-docker
Hi, I have just tested it and I am running in the same error someone else already mentioned.

![image](https://github.com/martialblog/docker-limesurvey/assets/85015179/fb808ed6-4f4c-477a-8e80-94e36ada49e0)
Used the latest image from adam

**Comment by bovender** [comment]
Well that's odd. I've had some very "interesting" issues with stray/dangling Docker volumes in the past, maybe you want to double-check that you _really_ start your container from scratch? Just an idea. Or try my `docker-compose.yml` file that I posted [above](#issuecomment-1740588958) and see if that makes a difference or not.

**Comment by jonas-kiesewetter** [comment]
Used your docker-compose-yml and facing the issue for martialblog
from above:
> ![image](https://private-user-images.githubusercontent.com/54589115/330686086-fd0def71-713f-48dd-ab08-6475b9da1817.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTkzODQxMzMsIm5iZiI6MTcxOTM4MzgzMywicGF0aCI6Ii81NDU4OTExNS8zMzA2ODYwODYtZmQwZGVmNzEtNzEzZi00OGRkLWFiMDgtNjQ3NWI5ZGExODE3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjI2VDA2MzcxM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY4NjFiNTM5NzlmNzgwYTk4YjU1Y2U0ZDg0MjYxNGQ5OWRmMzk4M2EwYWMzODIwYjc2Yzk4Yzk2MGFlMWY4ZmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.G2B8jmKA3MR58MK9-bcu6tHzXGnMkcWtjpdzqEPScAQ)

and the one I have provided yesterday

**Comment by bovender** [comment]
@jonas-kiesewetter That's too bad, I'm afraid I do not have enough Docker karma to solve this... The `docker-compose.yml` file still works well on my end (as does my production system), so it makes me wonder whether there is something specific to your server or workstation (Docker version???) that interferes with it. Do you have other hardware to try this on, by any chance?

**Comment by Chektal** [comment]
Hello,

I had the same problem but on both docker images.
I managed to solve the problem by mapping the following volumes:
- ./config.php:/var/www/html/application/config/config.php
In the config.php file, comment out the 'baseUrl' line as below:
// ‘baseUrl’ => ‘ ’,
- ./kcfinder.php:/var/www/html/vendor/kcfinder/config/config.php
In the file, add your server's URL in 'cookieDomain' as in the example below
 'cookieDomain' => "mywebsite.fr",

**Comment by drdiscipulus** [comment]
I ran into the same problem with image uploads in the survey Resources tab.

The Resources/KCFinder window opens, but upload/refresh/files do not work correctly. After debugging it, it looks like KCFinder loads its JavaScript and CSS from the wrong path.

KCFinder is served from `/vendor/kcfinder/`, but this file:

`/var/www/html/vendor/kcfinder/tpl/tpl_browser.php`

contains this base tag:

`<base href="<?= Yii::app()->request->getBaseUrl() ?>/">`

Because of that, the browser tries to load KCFinder assets from paths like `/js/index.php`, `/css/index.php`, and `/themes/default/js.php` instead of `/vendor/kcfinder/js/index.php`, `/vendor/kcfinder/css/index.php`, and `/vendor/kcfinder/themes/default/js.php`.

So the UI is visible, but KCFinder's JavaScript is broken or incomplete.

I was able to fix it, at least for my setup, by changing the base tag to:

`<base href="<?= Yii::app()->getConfig("publicurl") ?>/vendor/kcfinder/">`

After that, uploads and refresh worked in the Resources tab, and files were correctly saved under:

`/var/www/html/upload/surveys/<surveyid>/images/`

and:

`/var/www/html/upload/surveys/<surveyid>/files/`

This image writes the Docker `BASE_URL` value into LimeSurvey's `request.baseUrl` config. Adam Zammit's image does not seem to do that in the same way, and this seems to be the key difference. In this image, it makes KCFinder generate the wrong `<base href>`, so JS/CSS are loaded from the LimeSurvey root instead of `/vendor/kcfinder/`, which breaks the Resources tab upload UI.

**Comment by jonas-kiesewetter** [comment]
@drdiscipulus 
Hab mir grad dein Profil angeschaut, finde deinen usernmane hier richtig richtig cool gewählt. Wollt ich nur gesagt haben :D

**Comment by drdiscipulus** [comment]
@jonas-kiesewetter 
Thanks! I basically forked my dad’s username and added two letters to it. Having a funny username is about the only thing Latin at school turned out to be useful for, at least in my case

**Comment by martialblog** [comment]
@drdiscipulus Thanks for the detailed information. I wouldn't want to override the `tpl_browser.php` since it's a file provided by the application, is there any other way we can fix this?

For example, what happens when we simply don't set the `request.baseUrl`? We could skip that in the entrypoint when the BASE_URL variable is not set

**Comment by drdiscipulus** [comment]
@martialblog Your suggestion seems to work:

- Keeping `request.baseUrl` with a full URL breaks KCFinder.
- Setting `request.baseUrl` to an empty string also still breaks KCFinder.
- Removing the `request.baseUrl` config key entirely lets Yii auto-detect the correct base path for KCFinder.

With no `request.baseUrl` entry in `application/config/config.php`, the original KCFinder template:

`<base href="<?= Yii::app()->request->getBaseUrl() ?>/">`

generates:

`<base href="/vendor/kcfinder/">`

Then KCFinder loads its assets correctly.

I verified locally that, with the original `tpl_browser.php` restored and `request.baseUrl` removed. So I agree that the better image-level fix is probably in the entrypoint.

**Comment by martialblog** [comment]
Thanks for testing this and the feedback! Then we will update the entrypoint to not set the `request.baseUrl` then we BASE_URL env variable is not set explicitly. 

I'll try to work on this the next days. If someone else wants to open a PR that is also very welcome.

**Comment by drdiscipulus** [comment]
@martialblog 

I gave the pull request a try, but this was my first time working with GitHub in this way, so I hope I did everything more or less correctly

**Comment by martialblog** [comment]
@drdiscipulus Thanks for the work. I've merged the PR and once a new LS version is released the changes are in the image.
